spec/lib/vmdb/loggers_spec.rb

```diff
@@ -0,0 +1,52 @@
+describe Vmdb::Loggers do
+  let(:log_file) { Rails.root.join("log", "foo.log").to_s }
+  describe "#create_multicast_logger" do
+    it "defaults the lower level loggers to `DEBUG`" do
+      log = described_class.send(:create_multicast_logger, log_file)
+
+      expect(log.loggers.first.level).to eq(Logger::DEBUG)
+      expect(log.loggers.to_a.size).to   eq(1)
+    end
+
+    context "in a container environment" do
+      around(:each) { |example| in_container_env(example) }
+
+      it "sets logger_instance and $container_log to debug" do
+        log = described_class.send(:create_multicast_logger, log_file)
+
+        expect(log.loggers.first.level).to     eq(Logger::DEBUG)
+        expect(log.loggers.to_a.last.level).to eq(Logger::DEBUG)
+      end
+    end
+  end
+  describe "#apply_config_value" do
+    it "will update the main lower level logger instance" do
+      log = described_class.send(:create_multicast_logger, log_file)
+      described_class.send(:apply_config_value, {:foo => :info}, log, :foo)
+
+      expect(log.loggers.first.level).to eq(Logger::INFO)
+      expect(log.loggers.to_a.size).to   eq(1)
+    end
+
+    context "in a container environment" do
+      around(:each) { |example| in_container_env(example) }
+
+      it "will always keep $container_log as DEBUG" do
+        log = described_class.send(:create_multicast_logger, log_file)
+        described_class.send(:apply_config_value, {:foo => :info}, log, :foo)
+
+        expect(log.loggers.first.level).to     eq(Logger::INFO)
+        expect(log.loggers.to_a.last.level).to eq(Logger::DEBUG)
+      end
+    end
+  end
+end
+
+def in_container_env(example)
+  old_env = ENV.delete('CONTAINER')
+  ENV['CONTAINER'] = 'true'
+  example.run
+ensure
+  # ENV['x'] = nil deletes the key because ENV accepts only string values
```

Comment:
Not sure why this comment is needed...we should always set back to the original value anyway (i.e. this code won't ever say `= nil`, so no need to explain it)

Reply:
> this code won't ever say = `nil`, so no need to explain it

Not directly, but in the lines above, where we get the `old_env`, more often then not, it is probably `nil`.

Also, I have shamelessly ripped this off from here:  https://github.com/ManageIQ/manageiq/commit/d098bc0

So this is not my comment to begin with.

That said, I am fine with removing it if it is a bother, though, seeing as it is now in our codebase about 5 other times, we probably should just make it a helper at this point (not in this PR though).